package/MDAnalysis/analysis/MSD.py

```diff
@@ -0,0 +1,398 @@
+        entry, organized:
+        selection X tau
+
+    Methods
+    -------
+    _select(self)
+        Generates list of atom groups that satisfy select_list strings
+    _init_pos_deque(self):
+        Initializes lists necessary for MSD calculations at the first restart.
+    _update_deques(self, this_restart):
+        Updates lists necessary for MSD calculations
+    _process_pos_data(self):
+        Runs MSD calculation
+    run(self):
+        Calls _process_pos_data and saves final output
+
+    """
+
+    def __init__(self, universe, select_list, start_frame, stop_frame,
+                 dt_restart, out_name='msd', len_msd=250, write_output=True,
+                 max_data=False):
+        """
+        Parameters
+        ----------
+        universe : object
+            Universe object containing trajectory data
+        select_list : list
+            List of selection strings
+        start_frame : int
+            Time when analysis begins (in frames)
+        stop_frame : int
+            Time when analysis ends (in frames)
+        dt_restart : int
+            Spacing of restarts (in frames)
+        out_name: str
+            Name of pickle output file: (out_name).p
+        len_msd : int
+            Specifies the maximum tau to be calculated (frames)
+            (saves memory in array allocation)
+        write_output : bool
+            Specifies whether to write pickle file
+        max_data : bool
+            If true, include tau values after stop_frame in calculation (this
+            is useful for avoiding data loss when breaking large calculations
+            across multiple batch jobs)
+        """
+
+        self.universe = universe
+        self.select_list = select_list
+        if isinstance(select_list, str):
+            self.select_list = select_list.split(',')
+        print(self.select_list)
+        self.start_frame = int(start_frame)
+        self.stop_frame = int(stop_frame)
+        self.dt_restart = int(dt_restart)
+        self.out_name = out_name
+        # handling short simulation cases
+        n_frames = int(stop_frame) - int(start_frame) + 1
+        if int(len_msd) > int(n_frames):
+            len_msd = int(n_frames)
+        self.len_msd = int(len_msd)
+        self.write_output = write_output
+        self.max_data = max_data
+
+        # initializing object attributes
+        self.pos, self.atomids_per_sel_at_t, self.map_atomids_to_pos_at_t, = \
+            self._init_pos_deque()
+
+        n_sel = len(self.pos)
+        self.msd = np.zeros((n_sel, self.len_msd), dtype='float64')
+        self.n_samples = np.zeros((n_sel, self.len_msd), dtype='int')
+
+        # check that trajectory length is correct
+        try:
+            assert len(self.universe.trajectory) >= \
+                    (self.stop_frame-self.start_frame+1)
+        except RuntimeError:
+            print("Sample interval exceeds trajectory length. This may result"
```

Comment:
Please raise Exceptions to stop execution. But as mentioned above another convenience function can do this check for you.